src/plugins/registration.go

```diff
@@ -153,23 +156,43 @@ func (r *OneTimeRegistration) areDataplaneSoftwareDetailsReady() error {
 func (r *OneTimeRegistration) registerAgent() {
 	var details []*proto.NginxDetails
 
-	for _, proc := range r.processes {
-		// only need master process for registration
-		if proc.IsMaster {
-			nginxDetails := r.binary.GetNginxDetailsFromProcess(proc)
-			details = append(details, nginxDetails)
-			// Reading nginx config during registration to populate nginx fields like access/error logs, etc.
-			_, err := r.binary.ReadConfig(nginxDetails.GetConfPath(), nginxDetails.NginxId, r.env.GetSystemUUID())
-			if err != nil {
-				log.Warnf("Unable to read config for NGINX instance %s, %v", nginxDetails.NginxId, err)
+	backoffSetting := backoff.BackoffSettings{
+		InitialInterval: softwareDetailsOperationInterval,
+		MaxInterval:     softwareDetailsOperationInterval,
+		MaxElapsedTime:  nginxStartMaxWaitTime,
+		Jitter:          backoff.BACKOFF_JITTER,
+		Multiplier:      backoff.BACKOFF_MULTIPLIER,
+	}
+
+	findMaster := func() error {
+		for _, proc := range r.processes {
+			// only need master process for registration
+			if proc.IsMaster {
+				nginxDetails := r.binary.GetNginxDetailsFromProcess(proc)
+				details = append(details, nginxDetails)
+				if len(details) > 0 {
+					// Reading nginx config during registration to populate nginx fields like access/error logs, etc.
+					_, err := r.binary.ReadConfig(nginxDetails.GetConfPath(), nginxDetails.NginxId, r.env.GetSystemUUID())
+					if err != nil {
+						log.Warnf("Unable to read config for NGINX instance %s, %v", nginxDetails.NginxId, err)
+					}
+					log.Info("Master process has been found")
+					return nil
+				}
+			} else {
+				log.Tracef("NGINX non-master process: %d", proc.Pid)
+				return nil
 			}
-		} else {
-			log.Tracef("NGINX non-master process: %d", proc.Pid)
 		}
+		return fmt.Errorf("No master process found, waiting...")
 	}
-	if len(details) == 0 {
-		log.Info("No master process found")
+	err := backoff.WaitUntil(
+		context.Background(), backoffSetting, findMaster,
+	)
+	if err != nil {
+		log.Warn(err.Error())
```

Comment:
This was previously:
```
    log.Info("No master process found")
```
But my instinct is that this should be an error log, as I'd think we can't complete registration if we haven't found a master process.

Others could also weigh in on this, as I wouldn't consider myself an expert on this.

Reply:
Yes I believe previously we would continue with registration even if we didn't find master processes but now that we are waiting until we find one I think it should be an error log message.
```
log.Errorf("Unable to find NGINX master processes, %v", err)
```